pydra/mark/tests/test_shell.py

```diff
@@ -0,0 +1,467 @@
+
+    # Test cmdline
+    assert ls.inputs.directory == tmpdir
+    assert not ls.inputs.hidden
+    assert ls.inputs.long_format
+    assert ls.cmdline == f"ls -l {tmpdir}"
+
+    # Drop Long format flag to make output simpler
+    ls = Ls(directory=tmpdir)
+    result = ls()
+
+    assert result.output.entries == ["a", "b", "c"]
+
+
+@pytest.fixture(params=["static", "dynamic"])
+def A(request):
+    if request.param == "static":
+
+        @shell_task
+        class A:
+            executable = "cp"
+
+            class Inputs:
+                x: os.PathLike = shell_arg(
+                    help_string="an input file", argstr="", position=0
+                )
+                y: str = shell_arg(
+                    help_string="path of output file",
+                    output_file_template="{x}_out",
+                    argstr="",
+                )
+
+    elif request.param == "dynamic":
+        A = shell_task(
+            "A",
+            executable="cp",
+            input_fields={
+                "x": {
+                    "type": os.PathLike,
+                    "help_string": "an input file",
+                    "argstr": "",
+                    "position": 0,
+                },
+                "y": {
+                    "type": str,
+                    "help_string": "path of output file",
+                    "argstr": "",
+                    "output_file_template": "{x}_out",
+                },
+            },
+        )
+    else:
+        assert False
+
+    return A
+
+
+def test_shell_output_file_template(A):
+    assert "y" in [a.name for a in attrs.fields(A.Outputs)]
+
+
+def test_shell_output_field_name_static():
+    @shell_task
+    class A:
+        executable = "cp"
+
+        class Inputs:
+            x: os.PathLike = shell_arg(
+                help_string="an input file", argstr="", position=0
+            )
+            y: str = shell_arg(
+                help_string="path of output file",
+                output_file_template="{x}_out",
+                output_field_name="y_out",
+                argstr="",
+            )
+
+    assert "y_out" in [a.name for a in attrs.fields(A.Outputs)]
+
+
+def test_shell_output_field_name_dynamic():
+    A = shell_task(
+        "A",
+        executable="cp",
+        input_fields={
+            "x": {
+                "type": os.PathLike,
+                "help_string": "an input file",
+                "argstr": "",
+                "position": 0,
+            },
+            "y": {
+                "type": str,
+                "help_string": "path of output file",
+                "argstr": "",
+                "output_field_name": "y_out",
+                "output_file_template": "{x}_out",
+            },
+        },
+    )
+
+    assert "y_out" in [a.name for a in attrs.fields(A.Outputs)]
+
+
+def get_file_size(y: Path):
+    result = os.stat(y)
+    return result.st_size
+
+
+def test_shell_bases_dynamic(A, tmpdir):
+    B = shell_task(
```

Comment:
so I understand that this has the same `input_spec` and `executable`/`args` as `A`?

Could we also overwrite `executable` or `args`?